### DISCUSSION TITLE
How to register with maximal accuracy for the brainstem

### DISCUSSION THREAD

**Discussion by polivaoren** [ANTsX/ANTs#1648] [General]
Hello. My research focuses on the inferior colliculus. Hence, I need to register my T1 and other images in a manner that minimizes deformations at this ROI. I was told that there is a tool for this kind of procedure with ANTs. I would highly appreciate more information and an example how it can be done. Sincerely, Oren.

**Comment by ntustison** [comment]
I would start with the [ANTsX tutorial](https://gist.github.com/ntustison/12a656a5fc2f6f9c4494c88dc09c5621) where you can try self-contained examples illustrating ANTs functionality through the R or Python interfaces.

**Comment by polivaoren** [comment]
Sorry, maybe I should have paraphrased better. I know how to use ANTs. May question is whether there is a specific protocol to minimize deformation of a specific ROI during the registration process?

**Comment by ntustison** [comment]
One can specify a mask but that only excludes that region for similarity measure calculation.   There is no masking option which explicitly minimizes the deformation within that masked region.  You might be able to play with the option of adding additional images of just that region to help decrease the amount of deformation for that region but that's application-specific and something you'd have to play with.  We don't have a specific protocol.

**Comment by cookpa** [comment]
If you have the ROI labeled in both spaces you could use a point set metric as a secondary metric

**Comment by polivaoren** [comment]
Can you show me an example how would this code look like

**Reply by cookpa** [reply]
Based on this example,

https://github.com/stnava/chicken/tree/master

You would add something like `-m PSE[${fixedPoints},${movingPoints},1]` where fixedPoints and movingPoints are the binary segmentation with your ROI. 

I'd add this after your main image metric - the first metric is used to set some optimization parameters and if the point set only covers a small ROI, it's best to set the image metric first.

Your antsRegistration call works in stages, with metric(s) and a transform for each. So you could do a rigid alignment with just the image metric, then add the point set for the affine and deformable stages.
